cmake/ThirdParty.cmake

```diff
@@ -180,6 +180,31 @@ include_directories(SYSTEM
   ${CMAKE_CURRENT_BINARY_DIR}/3rdparty/cli11-prefix/include
 )
 
+#google test
+include(ExternalProject)
+
+ExternalProject_Add(
+    googletest
+    SOURCE_DIR "${CMAKE_BINARY_DIR}/googletest-src"
+    BINARY_DIR "${CMAKE_BINARY_DIR}/googletest-build"
+    GIT_REPOSITORY https://github.com/google/googletest.git
+    GIT_TAG release-1.11.0
+    GIT_SHALLOW ON
+    INSTALL_COMMAND ""
+    TEST_COMMAND ""
+    LOG_DOWNLOAD ON
+    LOG_BUILD ON
+)
+
+option(BUILD_TESTING "Build tests" ON)
```

Comment:
Other options are defined in root CMakeLists.txt, so it's better to move it there too.